logbook-core/src/main/java/org/zalando/logbook/ExtendedLogFormatSink.java

```diff
@@ -0,0 +1,245 @@
+package org.zalando.logbook;
+
+import org.apiguardian.api.API;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+/**
+ * @see <a href="https://en.wikipedia.org/wiki/Extended_Log_Format">Wikipedia: Extended Log Format</a>
+ * @see <a href="https://www.w3.org/TR/WD-logfile.html">W3C Extended Log Format</a>
+ */
+@API(status = EXPERIMENTAL)
+public final class ExtendedLogFormatSink implements Sink {
+
+    private static final String DELIMITER = " ";
+    private static final String HEADER_DELIMITER = ";";
+    private static final String OMITTED_FIELD = "-";
+    private static final String DEFAULT_VERSION = "1.0";
+    private static final String DEFAULT_FIELDS = "date time c-ip s-dns cs-method cs-uri-stem cs-uri-query sc-status sc-bytes cs-bytes time-taken cs-protocol cs(User-Agent) cs(Cookie) cs(Referrer)";
+    private static final Pattern CS_HEADER_REGEX = Pattern.compile("cs\\((.*?)\\)");
+    private static final Pattern SC_HEADER_REGEX = Pattern.compile("sc\\((.*?)\\)");
+
+    private final HttpLogWriter writer;
+    private final ZoneId timeZone;
+    private final List<String> supportedFields;
+    private final List<String> fields;
+    private final DateTimeFormatter dateFormatter = DateTimeFormatter.ISO_LOCAL_DATE;
+    private final DateTimeFormatter timeFormatter = DateTimeFormatter.ISO_LOCAL_TIME;
+
+    public ExtendedLogFormatSink(final HttpLogWriter writer) {
+        this(writer, ZoneId.of("UTC"), DEFAULT_VERSION, DEFAULT_FIELDS);
+    }
+
+    public ExtendedLogFormatSink(final HttpLogWriter writer, final String fields) {
+        this(writer, ZoneId.of("UTC"), DEFAULT_VERSION, fields);
+    }
+
+    public ExtendedLogFormatSink(final HttpLogWriter writer, ZoneId timeZone, final String version,
+                                 final String fields) {
+        this.writer = writer;
+        this.timeZone = timeZone;
+        this.supportedFields = getSupportedFields();
+        this.fields = getFields(fields);
+        logDirectives(version, this.fields);
+    }
+
+    @Override
+    public boolean isActive() {
+        return writer.isActive();
+    }
+
+    @Override
+    public void write(final Precorrelation precorrelation, final HttpRequest request) {
+        // nothing to do...
```

Comment:
Out of curiosity, why do we do nothing here?

Reply:
Because this method is called upon request, but the format may include response fields which means all logging needs to be delayed until the response is received. Which is what the other `write` method is for.